src/main/scala/exercises/05_Patterns.scala

```diff
@@ -55,17 +59,21 @@ object PatternMatching {
    *        
    *    - anything else, the function result is "Some Scala class"
    */
-  def matchOnInputType(in: Any): String = {
-    error("fix me")
+  def matchOnInputType(in: Any): String = in match {
+    case s : String                  ⇒ s"A string with length ${s.length}"
+    case n : Int if n > 0            ⇒ "A positive integer"
+    case p : Person                  ⇒ s"A person with name: ${p.name}"
+    case s : Seq[_] if s.length > 10 ⇒ "Seq with more than 10 elements"
+    case first :: second :: tail     ⇒ s"first: $first, second: $second, rest: $tail"
+    case null                        ⇒ "A null value"
+    case _                           ⇒ "Some Scala class"
   }
 
```

Comment:
I really like this example because it shows the power of Scala pattern-matching.  In particular, the `case first :: second :: tail` is a very clean and clear way of both checking the list's length and extracting the elements.  Also, the string interpolation is quite readable, much more so than this code would be in a lower-level language like C++.